wandb/lib/config_util.py

```diff
@@ -43,17 +47,22 @@ def save_config_file_from_dict(config_filename, config_dict):
         conf_file.write(data)
 
 
-def dict_from_config_file(config_filename):
+def dict_from_config_file(filename, must_exist=False):
+    if not os.path.exists(filename):
+        if must_exist:
+            raise ConfigError("config file %s doesn't exist" % filename)
+        logger.debug('no default config file found in %s' % filename)
+        return None
     try:
-        conf_file = open(config_filename)
+        conf_file = open(filename)
     except OSError:
-        raise ConfigError("Couldn't read config file: %s" % config_filename)
+        raise ConfigError("Couldn't read config file: %s" % filename)
     try:
         loaded = load_yaml(conf_file)
     except yaml.parser.ParserError:
         raise ConfigError("Invalid YAML in config yaml")
     config_version = loaded.pop("wandb_version", None)
-    if config_version != 1:
+    if config_version and config_version != 1:
```

Comment:
`if config_version is not None and config_version != 1` (so that we raise if `config_version` is 0 as well)

Reply:
Thank you for all the feedback!!!